aesd-char-driver/main.c

```diff
@@ -12,220 +12,319 @@
  */
 
 #include <linux/module.h>
+#include <linux/syscalls.h>
 #include <linux/init.h>
 #include <linux/printk.h>
 #include <linux/types.h>
 #include <linux/cdev.h>
 #include <linux/fs.h> // file_operations
-#include "aesdchar.h"
-#include <linux/syscalls.h>
 #include <asm/uaccess.h>
+#include "aesdchar.h"
+#include "aesd_ioctl.h"
 
 int aesd_major =   0; // use dynamic major
 int aesd_minor =   0;
 
-MODULE_AUTHOR("Pranav Bharadwaj"); /** TODO: fill in your name **/
+MODULE_AUTHOR("Pranav Bharadwaj");
 MODULE_LICENSE("Dual BSD/GPL");
 
 struct aesd_dev aesd_device;
 
 int aesd_open(struct inode *inode, struct file *filp)
 {
-    PDEBUG("open");
-    /**
-     * TODO: handle open
-     */
-    //Increment device use count by a process
+    //Handle open
+    PDEBUG("open\n");
     try_module_get(THIS_MODULE);
-     
+
     return 0;
 }
 
 int aesd_release(struct inode *inode, struct file *filp)
 {
-    PDEBUG("release");
-    /**
-     * TODO: handle release
-     */
-    //Decrement device use count by a process
-    module_put(THIS_MODULE); 
-    
+    //Handle release
+    PDEBUG("release\n");
+    module_put(THIS_MODULE);
```

Comment:
@prvbharadwaj1731 see pull review comments from Assignment 8, this shouldn't be necessary.